src/gg20/crypto_tools/paillier/zk/composite_dlog.rs

```diff
@@ -64,8 +85,8 @@ fn compute_challenge(stmt: &CompositeDLogStmt, domain: &[u8], x: &BigNumber) ->
 fn legendre_symbol(a: &BigNumber, p: &BigNumber) -> i32 {
     debug_assert!(a.gcd(p).is_one());
 
-    let e = (p - 1) >> 1;
-    if a.modpow(&e, p).is_one() {
+    let e = Randomness((p - 1) >> 1);
```

Comment:
This is to `zeroize` on `drop`.

Reply:
I'm not a huge fan of this.  It's not clear why you're using `Randomness` here.  Perhaps your github PR comment should instead be a code comment. 😛 A proper solution would make the purpose clear.  (Sigh)  Do we need something like `Randomness` except it's called `ZeroizeOnDrop`?

Reply:
Could use the Secret crate from rustcrypto